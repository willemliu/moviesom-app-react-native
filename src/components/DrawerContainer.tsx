import React from 'react';
import { Text, View, AsyncStorage, Linking, TouchableNativeFeedback, Platform } from 'react-native';
import { NavigationActions } from 'react-navigation';
import {textStyle, viewStyle} from "../styles/Styles";
import TouchTextButton from './TouchTextButton';

export default class DrawerContainer extends React.Component<any, any> {
    props: any;
    state: any;

    constructor(props: any) {
        super(props);
        this.state = {
            loggedIn: this.props.navigation.getParam('loggedIn', null)
        };
        this.checkLogin();
    }

    checkLogin = async () => {
        this.setState({
            loggedIn: await AsyncStorage.getItem('loggedIn')
        });
    }

    logOut = async () => {
        await AsyncStorage.removeItem('loggedIn');
        this.setState({
            loggedIn: null
        });

        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Drawer' })],
        });
        this.props.navigation.dispatch(resetAction);
    }

    render() {
        const { navigation } = this.props;
        return (
            <View style={viewStyle.drawer}>
                <TouchTextButton style={{marginBottom: 10}} onPress={() => navigation.navigate('Home')}>Home</TouchTextButton>

                {this.state.loggedIn ? null : (
                    <TouchTextButton style={{marginBottom: 10}} onPress={() => navigation.navigate('Login')}>Login</TouchTextButton>
                )}

                {this.state.loggedIn ? null : (
                    <TouchTextButton style={{marginBottom: 10}} onPress={() => navigation.navigate('SignUp')}>Sign up</TouchTextButton>
                )}

                {this.state.loggedIn ? (
                    <TouchTextButton style={{marginBottom: 10}} onPress={this.logOut}>Log out</TouchTextButton>
                ) : null}

                <TouchTextButton
                    style={{marginBottom: 10}}
                    onPress={() => Linking.openURL('exp://exp.host/@willem_liu/react-native-ts?tmdbMovieId=500')}
                >Link external</TouchTextButton>
                <TouchTextButton
                    style={{marginBottom: 10}}
                    onPress={() => this.props.navigation.navigate('Donate', {url: 'https://app.moviesom.com'})}
                >MovieSom</TouchTextButton>

                <TouchTextButton
                    style={{marginBottom: 10}}
                    onPress={() => navigation.navigate('About', {name: 'Willem Liu'})}
                >About</TouchTextButton>

                <TouchTextButton
                    style={{marginBottom: 10}}
                    onPress={async () => await AsyncStorage.removeItem('store')}
                >Clear DB</TouchTextButton>

            </View>
        );
    }
}
