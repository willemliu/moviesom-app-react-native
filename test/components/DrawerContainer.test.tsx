import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import DrawerContainer from "../../src/components/DrawerContainer";

describe('DrawerContainer', () => {
    it('should render correctly', () => {
        const rendered = renderer.create(<DrawerContainer onPress={jest.fn()}><Text>Touchable component</Text></DrawerContainer>).toJSON();
        expect(rendered).toMatchSnapshot();
    });
});
