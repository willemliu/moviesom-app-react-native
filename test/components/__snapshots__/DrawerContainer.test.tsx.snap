// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerContainer should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008CBA",
          "borderColor": "#008CBA",
          "borderRadius": 3,
          "borderWidth": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008CBA",
          "borderColor": "#008CBA",
          "borderRadius": 3,
          "borderWidth": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Login
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008CBA",
          "borderColor": "#008CBA",
          "borderRadius": 3,
          "borderWidth": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Sign up
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008CBA",
          "borderColor": "#008CBA",
          "borderRadius": 3,
          "borderWidth": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      About
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#008CBA",
          "borderColor": "#008CBA",
          "borderRadius": 3,
          "borderWidth": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Clear DB
    </Text>
  </View>
</View>
`;
